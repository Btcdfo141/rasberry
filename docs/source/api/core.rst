.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: