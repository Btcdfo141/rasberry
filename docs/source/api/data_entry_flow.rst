.. _data_entry_flow_module:

:mod:`spencerassistant.data_entry_flow`
-----------------------------

.. automodule:: spencerassistant.data_entry_flow
    :members:
