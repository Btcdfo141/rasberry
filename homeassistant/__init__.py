"""Init file for Home Assistant."""


test2
