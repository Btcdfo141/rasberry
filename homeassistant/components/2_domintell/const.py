"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
