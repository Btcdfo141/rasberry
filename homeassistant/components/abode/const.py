"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
