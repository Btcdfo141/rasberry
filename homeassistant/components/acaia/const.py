"""Constants for component."""
from typing import Final

DOMAIN: Final = "acaia"
CONF_IS_NEW_STYLE_SCALE: Final = "is_new_style_scale"
