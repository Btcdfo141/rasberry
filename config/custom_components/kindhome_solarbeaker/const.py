from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
