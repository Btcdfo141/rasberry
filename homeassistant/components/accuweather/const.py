"""Constants for AccuWeather integration."""
from __future__ import annotations

from typing import Final

from homeassistant.components.weather import (
    ATTR_CONDITION_CLEAR_NIGHT,
    ATTR_CONDITION_CLOUDY,
    ATTR_CONDITION_EXCEPTIONAL,
    ATTR_CONDITION_FOG,
    ATTR_CONDITION_HAIL,
    ATTR_CONDITION_LIGHTNING,
    ATTR_CONDITION_LIGHTNING_RAINY,
    ATTR_CONDITION_PARTLYCLOUDY,
    ATTR_CONDITION_POURING,
    ATTR_CONDITION_RAINY,
    ATTR_CONDITION_SNOWY,
    ATTR_CONDITION_SNOWY_RAINY,
    ATTR_CONDITION_SUNNY,
    ATTR_CONDITION_WINDY,
)
from homeassistant.const import (
    CONCENTRATION_PARTS_PER_CUBIC_METER,
    DEVICE_CLASS_TEMPERATURE,
    LENGTH_FEET,
    LENGTH_INCHES,
    LENGTH_METERS,
    LENGTH_MILLIMETERS,
    PERCENTAGE,
    SPEED_KILOMETERS_PER_HOUR,
    SPEED_MILES_PER_HOUR,
    TEMP_CELSIUS,
    TEMP_FAHRENHEIT,
    TIME_HOURS,
    UV_INDEX,
)

from .model import SensorDescription

ATTRIBUTION: Final = "Data provided by AccuWeather"
ATTR_FORECAST: Final = "forecast"
CONF_FORECAST: Final = "forecast"
COORDINATOR: Final = "coordinator"
DOMAIN: Final = "accuweather"
MANUFACTURER: Final = "AccuWeather, Inc."
NAME: Final = "AccuWeather"
UNDO_UPDATE_LISTENER: Final = "undo_update_listener"

CONDITION_CLASSES: Final[dict[str, list[int]]] = {
    ATTR_CONDITION_CLEAR_NIGHT: [33, 34, 37],
    ATTR_CONDITION_CLOUDY: [7, 8, 38],
    ATTR_CONDITION_EXCEPTIONAL: [24, 30, 31],
    ATTR_CONDITION_FOG: [11],
    ATTR_CONDITION_HAIL: [25],
    ATTR_CONDITION_LIGHTNING: [15],
    ATTR_CONDITION_LIGHTNING_RAINY: [16, 17, 41, 42],
    ATTR_CONDITION_PARTLYCLOUDY: [3, 4, 6, 35, 36],
    ATTR_CONDITION_POURING: [18],
    ATTR_CONDITION_RAINY: [12, 13, 14, 26, 39, 40],
    ATTR_CONDITION_SNOWY: [19, 20, 21, 22, 23, 43, 44],
    ATTR_CONDITION_SNOWY_RAINY: [29],
    ATTR_CONDITION_SUNNY: [1, 2, 5],
    ATTR_CONDITION_WINDY: [32],
}

FORECAST_DAYS: Final[list[int]] = [0, 1, 2, 3, 4]

FORECAST_SENSOR_TYPES: Final[dict[str, SensorDescription]] = {
    "CloudCoverDay": {
        "device_class": None,
        "icon": "mdi:weather-cloudy",
        "label": "Cloud Cover Day",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "CloudCoverNight": {
        "device_class": None,
        "icon": "mdi:weather-cloudy",
        "label": "Cloud Cover Night",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "Grass": {
        "device_class": None,
        "icon": "mdi:grass",
        "label": "Grass Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "HoursOfSun": {
        "device_class": None,
        "icon": "mdi:weather-partly-cloudy",
        "label": "Hours Of Sun",
        "unit_metric": TIME_HOURS,
        "unit_imperial": TIME_HOURS,
    },
    "Mold": {
        "device_class": None,
        "icon": "mdi:blur",
        "label": "Mold Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "Ozone": {
        "device_class": None,
        "icon": "mdi:vector-triangle",
        "label": "Ozone",
        "unit_metric": None,
        "unit_imperial": None,
    },
    "Ragweed": {
        "device_class": None,
        "icon": "mdi:sprout",
        "label": "Ragweed Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "RealFeelTemperatureMax": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Max",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureMin": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Min",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureShadeMax": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Shade Max",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureShadeMin": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Shade Min",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "ThunderstormProbabilityDay": {
        "device_class": None,
        "icon": "mdi:weather-lightning",
        "label": "Thunderstorm Probability Day",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "ThunderstormProbabilityNight": {
        "device_class": None,
        "icon": "mdi:weather-lightning",
        "label": "Thunderstorm Probability Night",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "Tree": {
        "device_class": None,
        "icon": "mdi:tree-outline",
        "label": "Tree Pollen",
        "unit_metric": CONCENTRATION_PARTS_PER_CUBIC_METER,
        "unit_imperial": CONCENTRATION_PARTS_PER_CUBIC_METER,
    },
    "UVIndex": {
        "device_class": None,
        "icon": "mdi:weather-sunny",
        "label": "UV Index",
        "unit_metric": UV_INDEX,
        "unit_imperial": UV_INDEX,
    },
    "WindGustDay": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Gust Day",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindGustNight": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Gust Night",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindDay": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Day",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindNight": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Night",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
}

OPTIONAL_SENSORS: Final[tuple[str, ...]] = (
    "ApparentTemperature",
    "CloudCover",
    "CloudCoverDay",
    "CloudCoverNight",
    "DewPoint",
    "Grass",
    "Mold",
    "Ozone",
    "Ragweed",
    "RealFeelTemperatureShade",
    "RealFeelTemperatureShadeMax",
    "RealFeelTemperatureShadeMin",
    "Tree",
    "WetBulbTemperature",
    "WindChillTemperature",
    "WindGust",
    "WindGustDay",
    "WindGustNight",
)

SENSOR_TYPES: Final[dict[str, SensorDescription]] = {
    "ApparentTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Apparent Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "Ceiling": {
        "device_class": None,
        "icon": "mdi:weather-fog",
        "label": "Cloud Ceiling",
        "unit_metric": LENGTH_METERS,
        "unit_imperial": LENGTH_FEET,
    },
    "CloudCover": {
        "device_class": None,
        "icon": "mdi:weather-cloudy",
        "label": "Cloud Cover",
        "unit_metric": PERCENTAGE,
        "unit_imperial": PERCENTAGE,
    },
    "DewPoint": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Dew Point",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "RealFeelTemperatureShade": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "RealFeel Temperature Shade",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "Precipitation": {
        "device_class": None,
        "icon": "mdi:weather-rainy",
        "label": "Precipitation",
        "unit_metric": LENGTH_MILLIMETERS,
        "unit_imperial": LENGTH_INCHES,
    },
    "PressureTendency": {
        "device_class": "accuweather__pressure_tendency",
        "icon": "mdi:gauge",
        "label": "Pressure Tendency",
        "unit_metric": None,
        "unit_imperial": None,
    },
    "UVIndex": {
        "device_class": None,
        "icon": "mdi:weather-sunny",
        "label": "UV Index",
        "unit_metric": UV_INDEX,
        "unit_imperial": UV_INDEX,
    },
    "WetBulbTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Wet Bulb Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "WindChillTemperature": {
        "device_class": DEVICE_CLASS_TEMPERATURE,
        "icon": None,
        "label": "Wind Chill Temperature",
        "unit_metric": TEMP_CELSIUS,
        "unit_imperial": TEMP_FAHRENHEIT,
    },
    "Wind": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
    "WindGust": {
        "device_class": None,
        "icon": "mdi:weather-windy",
        "label": "Wind Gust",
        "unit_metric": SPEED_KILOMETERS_PER_HOUR,
        "unit_imperial": SPEED_MILES_PER_HOUR,
    },
}
