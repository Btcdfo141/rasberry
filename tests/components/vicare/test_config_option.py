"""Test the ViCare config flow."""

from syrupy.assertion import SnapshotAssertion

from homeassistant.components.vicare.const import CONF_EXTENDED_API, DOMAIN
from homeassistant.const import CONF_CLIENT_ID, CONF_PASSWORD, CONF_USERNAME
from homeassistant.core import HomeAssistant
from homeassistant.data_entry_flow import FlowResultType

from tests.common import MockConfigEntry

VALID_CONFIG = {
    CONF_USERNAME: "foo@bar.com",
    CONF_PASSWORD: "1234",
    CONF_CLIENT_ID: "5678",
}

VALID_OPTIONS = {
    CONF_EXTENDED_API: False,
}


async def test_user_options(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
) -> None:
    """Test that the option step works."""
    mock_entry = MockConfigEntry(
        domain=DOMAIN,
        unique_id="ViCare",
        data=VALID_CONFIG,
        options=VALID_OPTIONS,
    )
    mock_entry.add_to_hass(hass)

    result = await hass.config_entries.options.async_init(mock_entry.entry_id)

    assert result["type"] == FlowResultType.FORM
    assert result["step_id"] == "init"
    assert result["errors"] == {}

    result = await hass.config_entries.options.async_configure(
        result["flow_id"],
        {CONF_EXTENDED_API: True},
    )
    await hass.async_block_till_done()

    assert result["type"] == FlowResultType.CREATE_ENTRY
    assert result["data"] == snapshot
