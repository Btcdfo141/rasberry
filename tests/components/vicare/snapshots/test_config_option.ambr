# serializer version: 1
# name: test_user_options
  dict({
    'extended_api': True,
  })
# ---
